packages/client/components/TransactionDetails.tsx

```diff
@@ -0,0 +1,255 @@
+                                </Pressable>
+                            )}
+                        </HStack>
+                    </Box>
+                </VStack>
+            </Box>
+        );
+    }
+    function renderTransactionFee() {
+        // TODO
+        // Get Gas
+        // Get Gas Limit
+        return (
+            <Box style={styles.itemWrapper}>
+                <VStack>
+                    <Box>
+                        <Text style={styles.categoryTitle}>Transaction Fee</Text>
+                    </Box>
+                    <Box style={styles.subItemWrapper}>
+                        <Text color="text.500" style={styles.elementInformationTitle}>
+                            Gas Price
+                        </Text>
+                        <Text>0.0000000000012323 ETH</Text>
+                    </Box>
+                    <Box style={styles.subItemWrapper}>
+                        <HStack>
+                            <Text>Gas Limit</Text>
+                            <Text color="text.500" style={styles.gasLimit}>
+                                100 000
+                            </Text>
+                        </HStack>
+                    </Box>
+                </VStack>
+            </Box>
+        );
+    }
+    function renderOtherDetails() {
+        // TODO
+        // Add block number, position in block and nonce
+        return (
+            <Box style={styles.itemWrapper}>
+                <VStack>
+                    <Box>
+                        <Text style={styles.categoryTitle}>Other Details</Text>
+                    </Box>
+                    <Box style={styles.subItemWrapper}>
+                        <HStack>
+                            <Text>Block Number</Text>
+                            <Text color="text.500" style={styles.gasLimit}>
+                                15
+                            </Text>
+                        </HStack>
+                    </Box>
+                    <Box style={styles.subItemWrapper}>
+                        <HStack>
+                            <Text>Position in Block</Text>
+                            <Text color="text.500" style={styles.gasLimit}>
+                                19
+                            </Text>
+                        </HStack>
+                    </Box>
+                    <Box style={styles.subItemWrapper}>
+                        <HStack>
+                            <Text>Nonce</Text>
+                            <Text color="text.500" style={styles.gasLimit}>
+                                109,764
+                            </Text>
+                        </HStack>
+                    </Box>
+                </VStack>
+            </Box>
+        );
+    }
+
+    return (
+        <ScrollView>
+            {renderHeader()}
+            <Box>
+                {renderBasicInfo()}
+                {renderTransactionFee()}
+                {renderOtherDetails()}
+            </Box>
```

Comment:
Then here you can just reference the variable containing the jsx directly, its slightly faster but in general just best practice to not use a function when its not needed. Also do you need the `Box` wrapper here? in general I think yur wrapping everything in a box but all these components already implement box so you can apply the styling to it directly, see the comments attached to the component code.

```suggestion
            {renderHeader}
            <Box>
                {renderBasicInfo}
                {renderTransactionFee}
                {renderOtherDetails}
            </Box>
```

Reply:
Could even change `Box` with `VStack` with space of 40 and reduce CSS